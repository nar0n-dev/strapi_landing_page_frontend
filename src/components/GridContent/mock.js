export default {
  title: 'O titulo',
  html: `<p>
  Himenaeos et est laoreet ante hac duis libero netus aliquet pellentesque. Habitant augue purus hendrerit sagittis. Parturient montes dis gravida nibh parturient luctus malesuada curae;. Ut lacus habitasse erat tempus enim maecenas cras justo justo cubilia donec quam. Montes dui metus inceptos netus lectus himenaeos faucibus dui facilisi inceptos? Aliquam, inceptos cubilia ac nibh. Varius id adipiscing nascetur netus suscipit enim rutrum felis aliquam scelerisque. Sit, consequat mattis nibh et. Malesuada fermentum justo, vehicula quis egestas. Neque aptent tortor risus non. Sed enim inceptos fermentum penatibus conubia laoreet congue. Vulputate erat nullam venenatis vehicula torquent per quam himenaeos lacinia justo maecenas molestie. Tempus elit nascetur nisl suscipit facilisis turpis. Hac urna nascetur accumsan pulvinar sed iaculis sociis pulvinar euismod blandit magnis accumsan? Senectus ante felis eget pellentesque torquent sociosqu metus quisque, libero magna faucibus diam. Suspendisse congue viverra velit, faucibus.
  </p>
  <p>
  Morbi orci vehicula <a href="seila">vitae duis quisque gravida</a> varius. Ornare parturient felis mus et porttitor bibendum ullamcorper felis facilisis etiam. Dictum cursus, egestas porttitor elementum? Sociis massa lacus purus habitasse justo class libero cras primis. Taciti sapien non suscipit himenaeos ligula varius malesuada sem sem. Ad dictum habitant rutrum ac convallis blandit, himenaeos interdum nascetur non. Diam quam lacinia arcu natoque porttitor ornare. Nisl erat diam mattis lacinia donec proin et quisque. Netus luctus erat morbi egestas rhoncus dignissim ultricies sollicitudin vehicula. Eget sociosqu nec neque torquent laoreet. Montes pulvinar pharetra tempor enim dolor parturient inceptos interdum aliquet. Mauris, ornare sodales vestibulum. Elementum sed penatibus rhoncus curabitur. Tellus risus egestas himenaeos diam cubilia.
  </p>`,
  background: false,
}