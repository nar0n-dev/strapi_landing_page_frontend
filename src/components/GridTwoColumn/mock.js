export default {
  title: 'GridTwoColumn',
  text: `Lorem, ipsum dolor sit amet consectetur adipisicing elit. Voluptatibus, nesciunt magnam expedita in, voluptate soluta aliquid blanditiis eum vel deleniti reiciendis earum optio voluptates. Voluptatem non nisi aliquam nam delectus!`,
  srcImg: 'assets/images/javascript.svg',
}