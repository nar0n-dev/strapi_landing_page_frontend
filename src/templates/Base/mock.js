import linksmock from '../../components/NavLinks/mock';

export default {
  links: linksmock,
  logoData: {
    text: 'Logo',
    link: '#',
  },
  footerHtml: '<p>teste de footer</p>'
}