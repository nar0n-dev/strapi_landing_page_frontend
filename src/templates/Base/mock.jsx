import linksmock from '../../components/NavLinks/mock';

import gridMock from '../../components/GridText/mock';
import { GridText } from '../../components/GridText';

export const mockBase = {
  children: (
    <>
      <GridText {...gridMock} background/>
      <GridText {...gridMock}/>
      <GridText {...gridMock} background/>
      <GridText {...gridMock}/>
    </>
  ),
  links: linksmock,
  logoData: {
    text: 'Logo',
    link: '#',
  },
  footerHtml: '<p>teste de footer</p>'
}