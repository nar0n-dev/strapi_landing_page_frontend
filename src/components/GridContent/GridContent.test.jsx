import { GridContent } from '.';
import { screen } from '@testing-library/react';
import { renderTheme } from '../../styles/render-theme';

import mock from './mock';

describe('<GridContent />', () => {
  test('Test A - should render grid content', () => {
    const {container} = renderTheme(<GridContent {...mock} />);
    expect(container).toMatchSnapshot();
  });

  test('Test B - should render grid content withou background', () => {
    const {container} = renderTheme(<GridContent {...mock} background={undefined} />);
    expect(container).toMatchSnapshot();
  });
});