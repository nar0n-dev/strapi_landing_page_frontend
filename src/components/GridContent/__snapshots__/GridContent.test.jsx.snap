// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GridContent /> Test A - should render grid content 1`] = `
.c1 {
  max-width: 120rem;
  margin: 0 auto;
  padding: 3.2rem;
  width: 100%;
}

.c0 {
  background: #ffffff;
  color: #0A1128;
  min-height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c3 {
  color: #0A1128;
  font-size: 6.4rem;
  text-transform: uppercase;
}

.c5 {
  font-size: 2.4rem;
}

.c2 {
  text-align: center;
  max-width: 58rem;
  margin: 0 auto;
}

.c4 {
  margin: 6.4rem 0;
}

@media (max-width:768px) {
  .c3 {
    font-size: 4.0rem;
  }
}

<div>
  <div
    class="c0"
    id=""
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <h2
          class="c3"
        >
          O titulo
        </h2>
        <div
          class="c4"
        >
          <div
            class="c5"
          >
            <p>
              
  Himenaeos et est laoreet ante hac duis libero netus aliquet pellentesque. Habitant augue purus hendrerit sagittis. Parturient montes dis gravida nibh parturient luctus malesuada curae;. Ut lacus habitasse erat tempus enim maecenas cras justo justo cubilia donec quam. Montes dui metus inceptos netus lectus himenaeos faucibus dui facilisi inceptos? Aliquam, inceptos cubilia ac nibh. Varius id adipiscing nascetur netus suscipit enim rutrum felis aliquam scelerisque. Sit, consequat mattis nibh et. Malesuada fermentum justo, vehicula quis egestas. Neque aptent tortor risus non. Sed enim inceptos fermentum penatibus conubia laoreet congue. Vulputate erat nullam venenatis vehicula torquent per quam himenaeos lacinia justo maecenas molestie. Tempus elit nascetur nisl suscipit facilisis turpis. Hac urna nascetur accumsan pulvinar sed iaculis sociis pulvinar euismod blandit magnis accumsan? Senectus ante felis eget pellentesque torquent sociosqu metus quisque, libero magna faucibus diam. Suspendisse congue viverra velit, faucibus.
  
            </p>
            
  
            <p>
              
  Morbi orci vehicula 
              <a
                href="seila"
              >
                vitae duis quisque gravida
              </a>
               varius. Ornare parturient felis mus et porttitor bibendum ullamcorper felis facilisis etiam. Dictum cursus, egestas porttitor elementum? Sociis massa lacus purus habitasse justo class libero cras primis. Taciti sapien non suscipit himenaeos ligula varius malesuada sem sem. Ad dictum habitant rutrum ac convallis blandit, himenaeos interdum nascetur non. Diam quam lacinia arcu natoque porttitor ornare. Nisl erat diam mattis lacinia donec proin et quisque. Netus luctus erat morbi egestas rhoncus dignissim ultricies sollicitudin vehicula. Eget sociosqu nec neque torquent laoreet. Montes pulvinar pharetra tempor enim dolor parturient inceptos interdum aliquet. Mauris, ornare sodales vestibulum. Elementum sed penatibus rhoncus curabitur. Tellus risus egestas himenaeos diam cubilia.
  
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<GridContent /> Test B - should render grid content withou background 1`] = `
.c1 {
  max-width: 120rem;
  margin: 0 auto;
  padding: 3.2rem;
  width: 100%;
}

.c0 {
  background: #ffffff;
  color: #0A1128;
  min-height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c3 {
  color: #0A1128;
  font-size: 6.4rem;
  text-transform: uppercase;
}

.c5 {
  font-size: 2.4rem;
}

.c2 {
  text-align: center;
  max-width: 58rem;
  margin: 0 auto;
}

.c4 {
  margin: 6.4rem 0;
}

@media (max-width:768px) {
  .c3 {
    font-size: 4.0rem;
  }
}

<div>
  <div
    class="c0"
    id=""
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <h2
          class="c3"
        >
          O titulo
        </h2>
        <div
          class="c4"
        >
          <div
            class="c5"
          >
            <p>
              
  Himenaeos et est laoreet ante hac duis libero netus aliquet pellentesque. Habitant augue purus hendrerit sagittis. Parturient montes dis gravida nibh parturient luctus malesuada curae;. Ut lacus habitasse erat tempus enim maecenas cras justo justo cubilia donec quam. Montes dui metus inceptos netus lectus himenaeos faucibus dui facilisi inceptos? Aliquam, inceptos cubilia ac nibh. Varius id adipiscing nascetur netus suscipit enim rutrum felis aliquam scelerisque. Sit, consequat mattis nibh et. Malesuada fermentum justo, vehicula quis egestas. Neque aptent tortor risus non. Sed enim inceptos fermentum penatibus conubia laoreet congue. Vulputate erat nullam venenatis vehicula torquent per quam himenaeos lacinia justo maecenas molestie. Tempus elit nascetur nisl suscipit facilisis turpis. Hac urna nascetur accumsan pulvinar sed iaculis sociis pulvinar euismod blandit magnis accumsan? Senectus ante felis eget pellentesque torquent sociosqu metus quisque, libero magna faucibus diam. Suspendisse congue viverra velit, faucibus.
  
            </p>
            
  
            <p>
              
  Morbi orci vehicula 
              <a
                href="seila"
              >
                vitae duis quisque gravida
              </a>
               varius. Ornare parturient felis mus et porttitor bibendum ullamcorper felis facilisis etiam. Dictum cursus, egestas porttitor elementum? Sociis massa lacus purus habitasse justo class libero cras primis. Taciti sapien non suscipit himenaeos ligula varius malesuada sem sem. Ad dictum habitant rutrum ac convallis blandit, himenaeos interdum nascetur non. Diam quam lacinia arcu natoque porttitor ornare. Nisl erat diam mattis lacinia donec proin et quisque. Netus luctus erat morbi egestas rhoncus dignissim ultricies sollicitudin vehicula. Eget sociosqu nec neque torquent laoreet. Montes pulvinar pharetra tempor enim dolor parturient inceptos interdum aliquet. Mauris, ornare sodales vestibulum. Elementum sed penatibus rhoncus curabitur. Tellus risus egestas himenaeos diam cubilia.
  
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
